docs/content/addons/custom-helm-config.md

```diff
@@ -0,0 +1,72 @@
++++
+title = "Custom Helm Chart Configuration"
+icon = "fa-solid fa-eye"
++++
+
+A default helm chart configuration is provided with a ConfigMap `default-helm-addons-config`.
+This ConfigMap contains helm chart URL and version for each addon.
+A a ConfigMap with customized addon configuration can be created and referenced in the `Cluster` object.
+
+The helm chart configuration for an addon included in the custom ConfigMap will be installed on the cluster.
+If configuration for an addon is not included in the the customized addon configuration ConfigMap,
+it will be defauled from the `default-helm-addons-config` configmap.
+
+The content of the ConfigMap must use below specification:
+
+- It should be immutable
+- The format of the addon chart configuration should be:
+
+```yaml
+  addon-name: |
+    ChartName: addon-chart-name
+    ChartVersion: 1.0.0
+    RepositoryURL: https://my-chart-repository.example.com/charts/
```

Comment:
I wonder if we can enforce this format somehow without needing to jump to a whole CRD :thinking: 

just a thought i wanted to note down

Reply:
as we discussed:
- The new CRD would look like HelmChartProxy and CAREN's job would be take data from this new CRD and set it in HelmChartProxy.
- We want to keep using default configuration`default-helm-chart-config` to make it easy for customers to not worry about chart retrival details and making mistakes.
- CRD will be additional API on top of CAREN API which we think might not worth to maintain or overkill at this moment.

We decided to have a structured types instead of map of string values in the configmap and expose it though Go external library for GO clients. We can document the type schema for the API client. 
I will create a follow up documentation and PR to demo this.